src/core_types.jl

```diff
@@ -37,43 +52,58 @@ end
 # Only works because parameters of all the rotations are of a similar form
 # Would need to be more sophisticated if we have arbitrary dimensions, etc
 @inline function Base.promote_op{R1 <: Rotation, R2 <: Rotation}(::Type{R1}, ::Type{R2})
+    size(R1) == size(R2) || throw(DimensionMismatch("cannot promote rotations of $(size(R1)[1]) and $(size(R2)[1]) dimensions"))
     if isleaftype(R1)
         return R1
     else
         return R1{eltype(R2)}
     end
 end
 
-@inline function Base.:(/)(r1::Rotation, r2::Rotation)
+@inline function Base.:/(r1::Rotation, r2::Rotation)
     r1 * inv(r2)
 end
 
-@inline function Base.:(\)(r1::Rotation, r2::Rotation)
+@inline function Base.:\(r1::Rotation, r2::Rotation)
     inv(r1) * r2
 end
 
 ################################################################################
 ################################################################################
 """
-    immutable RotMatrix{T} <: Rotation{T}
+    immutable RotMatrix{N,T} <: Rotation{N,T}
 
-A statically-sized, 3×3 unitary (orthogonal) matrix.
+A statically-sized, N×N unitary (orthogonal) matrix.
 
 Note: the orthonormality of the input matrix is *not* checked by the constructor.
 """
-immutable RotMatrix{T} <: Rotation{T} # which is <: AbstractMatrix{T}
-    mat::SMatrix{3, 3, T, 9} # The final parameter to SMatrix is the "length" of the matrix, 3 × 3 = 9
+immutable RotMatrix{N,T,L} <: Rotation{N,T} # which is <: AbstractMatrix{T}
+    mat::SMatrix{N, N, T, L} # The final parameter to SMatrix is the "length" of the matrix, 3 × 3 = 9
 end
 
-# These 2 functions (plus size) are enough to satisfy the entire StaticArrays interface:
-@inline (::Type{RM}){RM<:RotMatrix}(t::NTuple{9}) = RM(SMatrix{3,3}(t))
+# These functions (plus size) are enough to satisfy the entire StaticArrays interface:
+@inline (::Type{R}){R<:RotMatrix}(t::Tuple)  = error("No precise constructor found. Length of input was $(length(t)).")
```

Comment:
I would have guess *StaticArrays* similar definition would have kicked in here. Is that not working?